diesel/src/pg/metadata_lookup.rs

```diff
@@ -0,0 +1,32 @@
+use std::mem;
+
+use prelude::*;
+use super::{PgConnection, PgTypeMetadata};
+
+#[allow(missing_debug_implementations)]
+pub struct PgMetadataLookup {
+    conn: PgConnection,
+}
+
+impl PgMetadataLookup {
+    pub(crate) fn new(conn: &PgConnection) -> &Self {
+        unsafe { mem::transmute(conn) }
```

Comment:
Do you think it would be valuable to add a comment which points to `Path::new` from the standard library as a reference point?